packages/devtools_app/lib/src/shared/feature_flags.dart

```diff
@@ -36,84 +37,180 @@ void setEnableExperiments() {
   /// https://github.com/flutter/devtools/issues/4470
-  static bool networkSaveLoad = true;
+  static final networkSaveLoad = BooleanFeatureFlag(
+    name: 'networkSaveLoad',
+    enabled: true,
+  );
 
   /// Flag to enable DevTools extensions.
   ///
   /// TODO(https://github.com/flutter/devtools/issues/6443): remove this flag
   /// once extension support is added in g3.
-  static bool devToolsExtensions = isExternalBuild;
+  static final devToolsExtensions = BooleanFeatureFlag(
+    name: 'devToolsExtensions',
+    enabled: isExternalBuild,
+  );
 
   /// Flag to enable debugging via DAP.
   ///
   /// https://github.com/flutter/devtools/issues/6056
-  static bool dapDebugging = enableExperiments;
+  static final dapDebugging = BooleanFeatureFlag(
+    name: 'dapDebugging',
+    enabled: enableExperiments,
+  );
 
   /// Flag to enable the new Inspector panel.
   ///
   /// https://github.com/flutter/devtools/issues/7854
-  static bool inspectorV2 = true;
+  static final inspectorV2 = BooleanFeatureFlag(
+    name: 'inspectorV2',
+    enabled: true,
+  );
 
   /// Flag to enable the DevTools setting to opt-in to WASM.
   ///
   /// https://github.com/flutter/devtools/issues/7856
-  static bool wasmOptInSetting = true;
+  static final wasmOptInSetting = BooleanFeatureFlag(
+    name: 'wasmOptInSetting',
+    enabled: true,
+  );
 
   /// Flag to enable refactors in the Flutter Property Editor sidebar.
   ///
   /// https://github.com/flutter/devtools/issues/9214
-  static bool propertyEditorRefactors = true;
+  static final propertyEditorRefactors = BooleanFeatureFlag(
+    name: 'propertyEditorRefactors',
+    enabled: true,
+  );
 
   /// Stores a map of all the feature flags for debugging purposes.
   ///
   /// When adding a new flag, you are responsible for adding it to this map as
   /// well.
-  static final _allFlags = <String, bool>{
-    'memoryObserver': memoryObserver,
-    'memorySaveLoad': memorySaveLoad,
-    'networkDisconnectExperience': networkDisconnectExperience,
-    'networkSaveLoad': networkSaveLoad,
-    'dapDebugging': dapDebugging,
-    'inspectorV2': inspectorV2,
-    'wasmOptInSetting': wasmOptInSetting,
-    'propertyEditorRefactors': propertyEditorRefactors,
+  static final _booleanFlags = <BooleanFeatureFlag>{
+    memoryObserver,
+    memorySaveLoad,
+    networkSaveLoad,
+    devToolsExtensions,
+    dapDebugging,
+    inspectorV2,
+    wasmOptInSetting,
+    propertyEditorRefactors,
+  };
+
+  static final _flutterChannelFlags = <FlutterChannelFeatureFlag>{
+    // TODO(https://github.com/flutter/devtools/issues/9438): Add wasm flag.
   };
 
   /// A helper to print the status of all the feature flags.
-  static void debugPrintFeatureFlags() {
-    for (final entry in _allFlags.entries) {
-      _log.config('${entry.key}: ${entry.value}');
+  static void debugPrintFeatureFlags({ConnectedApp? connectedApp}) {
+    for (final entry in _booleanFlags) {
+      _log.config('${entry.name}: ${entry.isEnabled}');
```

Comment:
Consider adding a `toString()` implementation to `BooleanFeatureFlag` that outputs this format. Ditto for `FlutterChannelFeatureFlag`.